Make ipsec-tools compatible with musl

ManoftheSea opened the pull request:
based on changes at
http://git.alpinelinux.org/cgit/aports/plain/main/ipsec-tools/musl-cdefs.patch

Also removed references to "login" and "logout" functions, as musl utmp
doesn't.

Signed-off-by: Derek LaHousse <dlahouss@mtu.edu>

thess commented:
Please bump PKG_RELEASE
@nmeyerhans -- OK

ManoftheSea commented:
Package release bumped.

thess commented:
OK thanks. Now, one more time around the block -- Squash/rebase into a single commit, add signed-off-by tag and prefix your commit message with the package name. Example:

_ipsec-tools: make compatible with musl_

ManoftheSea commented:
Haha, I learned how to rebase and force update.  Yay me.  Thanks thess.

nmeyerhans commented on net/ipsec-tools/patches/009-musl-compat.patch:
```diff
@@ -0,0 +1,187 @@
+@@ -1,6 +1,6 @@
+ /*	$NetBSD: extern.h,v 1.5 2009/03/06 11:45:03 tteras Exp $	*/
+ 
+-
```
The cleanup of extraneous whitespace (here and elsewhere), while nice, doesn't really belong in this change.

ManoftheSea commented:
Hmm, I don't think I understand what you mean.  There is no whitespace cleanup...
In setkey/extern.h, I replaced a blank line with an #include directive, to pull in sys/cdefs.h.  In racoon/isakmp_cfg.c, I deleted a line of code and its following empty line, to not have a resulting 2 empty lines together.  But that's all the whitespaces changes that I know of.

nmeyerhans commented:
Yeah, sorry, I was looking at the side-by-side diff and saw a line being removed but didn't see the replacement. Never mind my previous comment.

I've verified that we can still build against all supported libc alternatives, so this looks good.

Thanks @ManoftheSea and thanks @thess for reviewing this.
